Feature/config only context

jcampbell opened the pull request.

jcampbell committed:
Check for true lossiness before warning

Signed-off-by: James Campbell <james.p.campbell@gmail.com>

jcampbell committed:
WIP improve support for ConfigOnlyDataContext

Signed-off-by: James Campbell <james.p.campbell@gmail.com>

pep8speaks commented:
Thank you for submitting to Great Expectations!



* In the file [`great_expectations/data_context/data_context.py`](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/great_expectations/data_context/data_context.py):

> [Line 283:121](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/great_expectations/data_context/data_context.py#L283): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (148 > 120 characters)











* In the file [`great_expectations/data_context/store/html_site_store.py`](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/great_expectations/data_context/store/html_site_store.py):

> [Line 141:5](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/great_expectations/data_context/store/html_site_store.py#L141): [E301](https://duckduckgo.com/?q=pep8%20E301) expected 1 blank line, found 0
> [Line 176:22](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/great_expectations/data_context/store/html_site_store.py#L176): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

* In the file [`great_expectations/data_context/store/store.py`](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/great_expectations/data_context/store/store.py):

> [Line 74:35](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/great_expectations/data_context/store/store.py#L74): [W601](https://duckduckgo.com/?q=pep8%20W601) .has_key() is deprecated, use 'in'























* In the file [`great_expectations/datasource/generator/s3_generator.py`](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/great_expectations/datasource/generator/s3_generator.py):

> [Line 38:121](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/great_expectations/datasource/generator/s3_generator.py#L38): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (123 > 120 characters)

































* In the file [`tests/data_context/store/test_expectations_store.py`](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_expectations_store.py):

> [Line 33:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_expectations_store.py#L33): [W391](https://duckduckgo.com/?q=pep8%20W391) blank line at end of file



* In the file [`tests/data_context/store/test_store_backends.py`](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_store_backends.py):

> [Line 54:20](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_store_backends.py#L54): [W601](https://duckduckgo.com/?q=pep8%20W601) .has_key() is deprecated, use 'in'
> [Line 55:20](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_store_backends.py#L55): [W601](https://duckduckgo.com/?q=pep8%20W601) .has_key() is deprecated, use 'in'
> [Line 56:20](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_store_backends.py#L56): [W601](https://duckduckgo.com/?q=pep8%20W601) .has_key() is deprecated, use 'in'
> [Line 57:20](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_store_backends.py#L57): [W601](https://duckduckgo.com/?q=pep8%20W601) .has_key() is deprecated, use 'in'
> [Line 217:46](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_store_backends.py#L217): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 217:121](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_store_backends.py#L217): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (126 > 120 characters)
> [Line 242:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_store_backends.py#L242): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 244:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/data_context/store/test_store_backends.py#L244): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace











* In the file [`tests/datasource/generator/test_passthrough_generator.py`](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/datasource/generator/test_passthrough_generator.py):

> [Line 9:121](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/datasource/generator/test_passthrough_generator.py#L9): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (138 > 120 characters)

* In the file [`tests/jupyter_ux/test_jupyter_ux.py`](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/jupyter_ux/test_jupyter_ux.py):

> [Line 48:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/jupyter_ux/test_jupyter_ux.py#L48): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 54:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/jupyter_ux/test_jupyter_ux.py#L54): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 55:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/jupyter_ux/test_jupyter_ux.py#L55): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 56:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/jupyter_ux/test_jupyter_ux.py#L56): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 157:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/jupyter_ux/test_jupyter_ux.py#L157): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 158:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/jupyter_ux/test_jupyter_ux.py#L158): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 165:1](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/jupyter_ux/test_jupyter_ux.py#L165): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace







* In the file [`tests/test_configs.py`](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py):

> [Line 17:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L17): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 17:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L17): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 36:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L36): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 36:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L36): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 53:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L53): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 53:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L53): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 71:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L71): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 71:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L71): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 82:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L82): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 82:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L82): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 96:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L96): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 96:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L96): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 110:32](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L110): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 110:34](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L110): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 121:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L121): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 121:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L121): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 134:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L134): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 134:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L134): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 147:32](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L147): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 147:34](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L147): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 157:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L157): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 157:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L157): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 169:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L169): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 169:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L169): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 179:28](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L179): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 179:30](https://github.com/great-expectations/great_expectations/blob/81a913684e1e1d2cb6095027fa1decd1536620e0/tests/test_configs.py#L179): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals



This linter comment has been updated based on the most recent commit.

##### Comment last updated at 2019-12-09 22:04:35 UTC

jcampbell committed:
Support configurable boto3 options in generator and stores (#901)

Signed-off-by: James Campbell <james.p.campbell@gmail.com>

jcampbell committed:
Add support for passthrough_generator and manual_generator

Signed-off-by: James Campbell <james.p.campbell@gmail.com>

jcampbell committed:
Update changelog for release

jcampbell committed:
Update changelog for develop

roblim committed:
Improve Data Docs with Static Assets Rendering (#892)

* Copy data docs static assets directory

* Move logic into page action card

* Add static assets directory

* Add jinja global var for static_assets_dir

* Move styles into static dir

* Change headers to h4

* Add fonts to static dir

* Add logos and favicon to static images dir

* Add ge logo to top navbar; make logo link to home

* Add favicon

* Update syles dir

* Add favicon

* Nix dickens photo from ge_info; open ge link in new tab

* Add logic to handle categorical distributions

* Add expect_column_quantile_values_to_be_between renderer

* Add quantile strings for Q1, Q3 and Median; Add to validation results

* Add observed value to kl_divergence validation result chart

* Create content_block_container template

* Update styles template path

* Update tests

* Add data_context argument

* Update tests

* Update kl_divergence observed value display

- display "None (-infinity, infinity, or NaN)" when actual kl-divergence is None
- nix strong tag

* Fix typos

* Only copy static assets dir tree if doesn’t exist

* Add static assets dir to manifest

* Fix linter errors

* Nix scaffolding of static assets in init flow

- move to site builder

* Move static assets copying to site builder; set custom styles dir in site builder

- nix data_context init arg from View, passing in custom styles dir instead

* Use relative paths for asset dirs

* Update tests

* Fix typo

* Fix typo

* Use new copy_static_assets() store method to copy static assets

* Add new “static_assets” store backend

- special store with no key_length or filepath_template
- enables use of variable-length keys

* Create new copy_static_assets method

- uses .set() method of store backend to write files

* Add newline

* Allow key_length to be None

* Allow values to be of type bytes

* Allow variable-length keys when no filepath template is set

* Handle writing of bytes

* Update test

* Add ability to skip dirs or files

* Use isistance to check type

* Set content_encoding and content_type

* Get file content_type and encoding using mimetypes.guess_type

* Update test and add test case using backend with no filepath template

* Only pass content encoding for string types

* Update test

* Handle py2

- try/except is to support py2, since both str and bytes objects pass above condition

abegong committed:
First revisions

abegong committed:
soften nots

abegong committed:
Fix indentation after headers

abegong committed:
Test comments

abegong committed:
Comment out aspirational stuff

abegong committed:
Text for pluggable and extensible

abegong committed:
Deprecate links to nowhere

abegong committed:
batteries included

abegong committed:
Update README.md

Co-Authored-By: Taylor Miller <Aylr@users.noreply.github.com>

abegong committed:
Update README.md

Co-Authored-By: Taylor Miller <Aylr@users.noreply.github.com>

abegong committed:
Update README.md

Co-Authored-By: Taylor Miller <Aylr@users.noreply.github.com>

eugmandel committed:
Added How do I start section

abegong committed:
Merge pull request #918 from great-expectations/eugene/readme_update_201912_0

Added How do I start section

abegong committed:
Move quickstart and add mdashes

abegong committed:
Move Quickstart up and deprecate How do I learn more

abegong committed:
Drop bad char

kyleaton committed:
New Pip, explainer gif and images (#920)

* new pip gif and screenshot

* star ask in readme

abegong committed:
Fix dead links

abegong committed:
Merge branch 'readme_update_201912' of github.com:great-expectations/great_expectations into readme_update_201912

abegong committed:
Different redshift logo

abegong committed:
Fix down with pipeline debt link

abegong committed:
Merge pull request #916 from great-expectations/readme_update_201912

Readme update 201912

jcampbell committed:
Refactor stores to more cleanly support InMemoryStores

Signed-off-by: James Campbell <james.p.campbell@gmail.com>

jcampbell committed:
Add Database-backed defaults for ValidationsStore and ExpectationsStore

Signed-off-by: James Campbell <james.p.campbell@gmail.com>

jcampbell committed:
Updates to support config-only datacontext

Signed-off-by: James Campbell <james.p.campbell@gmail.com>

jcampbell committed:
Merge latest changes from develop

Signed-off-by: James Campbell <james.p.campbell@gmail.com>

jcampbell committed:
Merge latest changes from feature/types

Signed-off-by: James Campbell <james.p.campbell@gmail.com>
